packages/backend-data/src/convert_js_resolvers.ts

```diff
@@ -0,0 +1,81 @@
+import { Construct } from 'constructs';
+import { AmplifyData } from '@aws-amplify/data-construct';
+import { readFileSync } from 'fs';
+import * as appsync from 'aws-cdk-lib/aws-appsync';
+// TODO: uncomment after data-schema-types change is published
+// import { JsResolver } from '@aws-amplify/data-schema-types';
+
+// Temporarily adding this type for draft PR. Will be imported from data-schema-types
+// once that package is updated with this new type
+export type JsResolver = {
+  typeName: 'Mutation' | 'Query' | 'Subscription';
+  fieldName: string;
+  handlers: {
+    dataSource: string;
+    entry: string;
+  }[];
+};
+
+/**
+ * Default top-level pipeline resolver template
+ */
+const topLevelResolver = `
+export function request(ctx) {
+  return {}
+}
+export function response(ctx) {
+  return ctx.prev.result
+}
+`;
+
+/**
+ * Converts JS Resolver definition emitted by data-schema into AppSync pipeline
+ * resolvers via the L1 construct.
+ */
+export const convertJsResolverDefinition = (
+  scope: Construct,
+  amplifyApi: AmplifyData,
+  jsResolvers: JsResolver[]
+): void => {
+  if (!jsResolvers || jsResolvers.length < 1) {
+    return;
+  }
+
+  jsResolvers.forEach((resolver) => {
+    const functions = resolver.handlers.map((handler, idx) => {
+      const fnName = `Fn_${resolver.typeName}_${resolver.fieldName}_${idx + 1}`;
+      const fn = new appsync.CfnFunctionConfiguration(scope, fnName, {
+        apiId: amplifyApi.apiId,
+        dataSourceName: handler.dataSource,
+        name: fnName,
+        // handler.entry is a resolved absolute path
+        code: readFileSync(handler.entry, 'utf-8'),
+        runtime: {
+          name: 'APPSYNC_JS',
+          runtimeVersion: '1.0.0',
+        },
+      });
+      fn.node.addDependency(amplifyApi);
```

Comment:
it should be okay to add these explicitly, but CFN should also be able to figure it out from the apiId reference in the resource config

Reply:
I was running into some sequencing issues without making the dep explicit, specifically, errors re: referencing nonexistent data sources. I think the apiId gets resolved before all the data sources are available.